### PR TITLE
feat: interchain token service deployment

### PR DESCRIPTION
**Pull request by Foivos**

### CODE REVIEW

File: packages/axelar-local-dev/test.js
Comment (npty) [inline packages/axelar-local-dev/test.js:1]: remove?

**Review by npty — COMMENTED** [review]
could you add some tests to make sure the deployment is working as expected?

File: packages/axelar-local-dev/src/Network.ts [hunk 1]
```diff
@@ -203,6 +225,54 @@ export class Network {
         return this.create3Deployer;
     }
 
+    async deployInterchainTokenService() {
+        logger.log(`Deploying the InterchainTokenService for ${this.name}... `);
+        const deploymentSalt = keccak256('0x1234');
+        const wallet = this.ownerWallet;
+        const tokenManagerDeployer = await deployContract(wallet, TokenManagerDeployer, [this.create3Deployer.address]);
+        const standardizedTokenLockUnlock = await deployContract(wallet, StandardizedTokenLockUnlock);
+        const standardizedTokenMintBurn = await deployContract(wallet, StandardizedTokenMintBurn);
```
Comment (npty) [inline packages/axelar-local-dev/src/Network.ts:234]: can you use `Promise.all` to some of contract deployments here (which don't depends on other deployment) to make it a bit more efficient?

File: packages/axelar-local-dev/src/Network.ts [hunk 2]
```diff
@@ -203,6 +225,54 @@ export class Network {
...
+        const params = defaultAbiCoder.encode(['string[]', 'string[]'], [[], []]);
+
+        const linkerRouterProxy = await deployContract(wallet, RemoteAddressValidatorProxy, [linkerRouterImpl.address, wallet.address, params]);
+        const linkerRouter = new Contract(linkerRouterProxy.address, linkerRouterImpl.interface, wallet);
+        
+        const tokenManagerImplementations = [
+            await deployContract(wallet, TokenManagerLockUnlock, [interchainTokenServiceAddress]),
+            await deployContract(wallet, TokenManagerMintBurn, [interchainTokenServiceAddress]),
+            await deployContract(wallet, TokenManagerLiquidityPool, [interchainTokenServiceAddress]),
+        ].map(contract => contract.address);
```
Comment (npty) [inline packages/axelar-local-dev/src/Network.ts:251]: promise.all here as well

File: packages/axelar-local-dev/src/its.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        const service = network.interchainTokenService;
+        await (await service.connect(wallet).deployCustomTokenManager(salt, tokenManagerType as BigNumberish, params)).wait();
+        const tokenId = await service.getCustomTokenId(wallet.address, salt);
+        const tokenManagerAddress = await service.getTokenManagerAddress(tokenId);
+        return TokenManagerFactory.connect(tokenManagerAddress, wallet);
+    }
+
+    network.its.deployCustomTokenManager.lockUnlock = async (salt: string, tokenAddress: string, operator: string = network.ownerWallet.address, wallet: Wallet = network.ownerWallet) => {
+        const params = await network.interchainTokenService.getParamsLockUnlock(operator, tokenAddress);
+        return await deployCustomTokenManager(salt, 0, params, wallet);
```
Comment (npty) [inline packages/axelar-local-dev/src/its.ts:52]: `await` is unnecessary for the return statement. can u remove it and others below?

File: packages/axelar-local-dev/src/its.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    deployRemoteCanonicalToken: (tokenAddress: string, destinationChain: string | Network, gasValue?: BigNumberish, wallet?: Wallet) => Promise<IStandardizedToken>;
+    deployCustomTokenManager: {
+        lockUnlock: (salt: string, tokenAddress: string, operator?: string, wallet?: Wallet) => Promise<ITokenManager>;
+        mintBurn: (salt: string, tokenAddress: string, operator?: string, wallet?: Wallet) => Promise<ITokenManager>;
+        liquidityPool: (salt: string, tokenAddress: string, liquidityPool: string, operator?: string, wallet?: Wallet) => Promise<ITokenManager>;
+    }
+}
+
+export async function setupITS(network: Network) {
+    network.its = {} as any;
```
Comment (npty) [inline packages/axelar-local-dev/src/its.ts:17]: can you typed it as `ITS` instead of any?

**Comment by npty** [comment]
force pushed here to resolve merge conflicts

**Comment by raress96** [comment]
Hello. Is there a timeline when this will get merged into `main` branch?

**Review by npty — APPROVED** [review]

**Pull request merged**